### PR TITLE
NetBSD cross compiling

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by Ericson2314 — APPROVED** [review]
Awesome!!

**Comment by Ericson2314** [comment]
Wooo! Staging and then it's a-go? We should get this on some their mailing list or whatever. It should be a great way to draw in more people to Nix.

**Review by pSub — APPROVED** [review]

**Comment by matthewbauer** [comment]
To avoid breaking anything I might wait until branching off for 19.03

**Comment by matthewbauer** [comment]
I also am trying to upstream the GCC patch here:

https://gcc.gnu.org/ml/gcc-patches/2019-02/msg00198.html

**Comment by Ericson2314** [comment]
Good luck with the upstream!

**Comment by Mic92** [comment]
Hopefully easier then getting stuff into the LLVM.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -23258,8 +23260,10 @@ in
 
   fts = if stdenv.hostPlatform.isMusl then netbsd.fts else null;
 
-  inherit (recurseIntoAttrs (callPackages ../os-specific/bsd { }))
-          netbsd;
+  netbsd = callPackages ../os-specific/bsd/netbsd {};
```
Comment (vcunat) [inline pkgs/top-level/all-packages.nix]: This removed `recurseIntoAttrs` and thus the packages aren't built by Hydra anymore: https://hydra.nixos.org/eval/1512985?filter=&compare=1512944&full=#tabs-removed  Was that intentional?

**Comment by vcunat** [comment]
The bigger problem is that netbsd-make won't build on Darwin anymore, breaking lots of packages, e.g. https://hydra.nixos.org/job/nixpkgs/trunk/getent.x86_64-darwin

**Comment by vcunat** [comment]
Well, `netbsd.make` won't build even on `x86_64-linux` on current nixpkgs master.

**Comment by matthewbauer** [comment]
Very weird. I’ll look into it

**Pull request merged**